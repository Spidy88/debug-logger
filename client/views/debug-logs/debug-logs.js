import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import LogRow from './log-row';
import LogControls from './log-controls';
import EmptyFeedback from '../../components/empty-feedback';
import ServerStatus from '../../components/server-status';
import {
    Pane,
    Table
} from 'evergreen-ui';

class DebugLogs extends Component {
    static propTypes = {
        logs: PropTypes.array,
        status: PropTypes.string
    };

    static defaultProps = {
        logs: []
    };

    constructor(props) {
        super(props);

        this.renderEmptyState = this.renderEmptyState.bind(this);
        this.renderServerStatus = this.renderServerStatus.bind(this);
        this.render = this.render.bind(this);
    }

    renderEmptyState() {
        if( this.props.logs.length ) {
            return null;
        }

        let iconProps = {
            icon: 'barber',
            message: 'Waiting for logs',
            subMessage: 'Kick back and relax'
        };

        // Render a waiting for logs feedback
        return (
            <EmptyFeedback {...iconProps} />
        );
    }

    renderServerStatus() {
        if( !this.props.status ) {
            return null;
        }

        return (
            <ServerStatus status={this.props.status} />
        );
    }

    render() {
        return (
            <React.Fragment>
                <Pane
                    background="tint2"
                    paddingX="0.75rem"
                    display="flex"
                    position="fixed"
                    width="100%"
                    top={0}
                    elevation={1}>

                    <LogControls />
                </Pane>

                <Box marginTop="calc(40px + 2rem)">
                    { this.renderServerStatus() }

                    { this.renderEmptyState() }

                    <Table>
                        <Table.Body>
                            {
                                this.props.logs.map((log) => (
                                    <LogRow key={log.messageId} log={log} />
                                ))
                            }
                        </Table.Body>
                    </Table>
                </Box>
            </React.Fragment>
        );
    }
}

export default DebugLogs;
