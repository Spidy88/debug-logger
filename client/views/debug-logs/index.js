import React, { Component } from 'react';
import PropTypes from 'prop-types';
import LogRow from './log-row';
import LogControls from './log-controls';
import {
    Pane,
    Table
} from 'evergreen-ui';

class DebugLogs extends Component {
    static propTypes = {
        logs: PropTypes.array
    };

    static defaultProps = {
        logs: []
    };

    constructor(props) {
        super(props);

        this.render = this.render.bind(this);
    }

    render() {
        return (
            <React.Fragment>
                <Pane
                    background="tint2"
                    paddingX="0.75rem"
                    display="flex"
                    position="fixed"
                    width="100%"
                    top={0}
                    elevation={1}>

                    <LogControls />
                </Pane>

                <Table>
                    <Table.Body>
                        {
                            this.props.logs.map((log) => (
                                <LogRow key={log.messageId} log={log} />
                            ))
                        }
                    </Table.Body>
                </Table>
            </React.Fragment>
        );
    }
}

export default DebugLogs;
