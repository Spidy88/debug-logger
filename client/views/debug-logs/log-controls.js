import React, { PureComponent } from 'react';
import Box from 'ui-box';

import {
    SearchInput,
    SegmentedControl
} from 'evergreen-ui';

const LiveOption = {
    label: 'Live',
    value: 'live'
};

const PauseOption = {
    label: 'Pause',
    value: 'pause'
};

class LogControls extends PureComponent {
    constructor(props) {
        super(props);

        this.render = this.render.bind(this);
    }

    render() {
        return (
            <React.Fragment>
                <Box marginX="0.75rem" marginY="1rem" background="white">
                    <SegmentedControl
                        options={[LiveOption, PauseOption]}
                        defaultValue={LiveOption.value}
                        height={40}
                        flexShrink={1}
                        flexGrow={0}
                        width="10rem" />
                </Box>

                <Box marginX="0.75rem" marginY="1rem" width="100%">
                    <SearchInput
                        height={40}
                        flexGrow={1}
                        width="100%"
                        placeholder="Type to search..." />
                </Box>
            </React.Fragment>
        );
    }
}

export default LogControls;