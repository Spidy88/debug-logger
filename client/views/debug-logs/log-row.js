import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import moment from 'moment';
import {
    Table,
    defaultTheme
} from 'evergreen-ui';

const eventTextProps = {
    fontWeight: 500,
    textAlign: 'left'
};

const dateTextProps = {
    textAlign: 'right'
};

class LogRow extends PureComponent {
    static propTypes = {
        log: PropTypes.shape({
            messageId: PropTypes.string.isRequired,
            sentAt: PropTypes.number.isRequired,
            event: PropTypes.string
        }).isRequired
    };

    constructor(props) {
        super(props);

        this.formattedLogDate = this.formattedLogDate.bind(this);
        this.render = this.render.bind(this);
    }

    formattedLogDate() {
        let { log } = this.props;
        return moment(log.sentAt).format('YYYY/MM/DD HH:mm:ss');
    }

    render() {
        let {
            log,
            ...props
        } = this.props;

        return (
            <Table.Row paddingX="0.75em" {...props}>
                <Table.Cell
                    width={40}
                    flexBasis="auto"
                    flex={0}
                    marginX="0.75em">

                    <Box
                        is="svg"
                        viewBox="0 0 24 24"
                        style={ { fill: defaultTheme.fills.solid.blue.backgroundColor } }
                        width={24}
                        height={24}>

                        <path d="M12 2c5.514 0 10 4.486 10 10s-4.486 10-10 10-10-4.486-10-10 4.486-10 10-10zm0-2c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm6.25 8.891l-1.421-1.409-6.105 6.218-3.078-2.937-1.396 1.436 4.5 4.319 7.5-7.627z"/>
                    </Box>
                </Table.Cell>

                <Table.TextCell
                    width="8em"
                    flexBasis="auto"
                    flex={0}
                    textTransform="uppercase"
                    fontWeight={300}
                    test-id="logType">

                    { log.type }
                </Table.TextCell>

                <Table.TextCell
                    textProps={eventTextProps}
                    test-id="logMessage">

                    { log.event || log.messageId }
                </Table.TextCell>

                <Table.TextCell
                    width="10em"
                    marginX="0.75em"
                    flexBasis="auto"
                    flex={0}
                    textProps={dateTextProps}
                    test-id="logDate">

                    { this.formattedLogDate() }
                </Table.TextCell>
            </Table.Row>
        );
    }
}

export default LogRow;