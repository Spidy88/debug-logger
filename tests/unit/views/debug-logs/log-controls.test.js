import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { LogControls } from '../../../../client/views/debug-logs';

describe('<LogControls>', () => {
    let testQuery;
    let testIsLive;
    let handleLive;
    let handlePause;
    let handleQueryChange;

    beforeEach(() => {
        testQuery = '';
        testIsLive = true;
        handleLive = jest.fn();
        handlePause = jest.fn();
        handleQueryChange = jest.fn();
    });

    it('should invoke pause callback when live is active', () => {
        const logControls = mount(
            <LogControls
                query={testQuery}
                isLive={testIsLive}
                onLiveSelected={handleLive}
                onPauseSelected={handlePause}
                onQueryChanged={handleQueryChange} />
        );

        const pauseButton = logControls.find('input[value="pause"]');
        pauseButton.simulate('change');

        expect(handleLive.mock.calls.length).toEqual(0);
        expect(handlePause.mock.calls.length).toEqual(1);
    });

    it('should invoke live callback when pause is active', () => {
        testIsLive = false;

        const logControls = mount(
            <LogControls
                query={testQuery}
                isLive={testIsLive}
                onLiveSelected={handleLive}
                onPauseSelected={handlePause}
                onQueryChanged={handleQueryChange} />
        );

        const liveButton = logControls.find('input[value="live"]');
        liveButton.simulate('change');

        expect(handleLive.mock.calls.length).toEqual(1);
        expect(handlePause.mock.calls.length).toEqual(0);
    });

    it('should not invoke live callback when live is active', () => {
        const logControls = mount(
            <LogControls
                query={testQuery}
                isLive={testIsLive}
                onLiveSelected={handleLive}
                onPauseSelected={handlePause}
                onQueryChanged={handleQueryChange} />
        );

        const liveButton = logControls.find('input[value="live"]');
        liveButton.simulate('change');

        expect(handleLive.mock.calls.length).toEqual(0);
        expect(handlePause.mock.calls.length).toEqual(0);

    });

    it('should not invoke pause callback when pause is active', () => {
        testIsLive = false;

        const logControls = mount(
            <LogControls
                query={testQuery}
                isLive={testIsLive}
                onLiveSelected={handleLive}
                onPauseSelected={handlePause}
                onQueryChanged={handleQueryChange} />
        );

        const pauseButton = logControls.find('input[value="pause"]');
        pauseButton.simulate('change');

        expect(handleLive.mock.calls.length).toEqual(0);
        expect(handlePause.mock.calls.length).toEqual(0);
    });

    it('should invoke query change callback when the search text changes', () => {
        const logControls = mount(
            <LogControls
                query={testQuery}
                isLive={testIsLive}
                onLiveSelected={handleLive}
                onPauseSelected={handlePause}
                onQueryChanged={handleQueryChange} />
        );

        const searchInput = logControls.find('input[test-id="logSearch"]').first();
        searchInput.simulate('change');

        expect(handleQueryChange.mock.calls.length).toEqual(1);
    });

    it('should match its snapshot', () => {
        const tree = renderer.create(
            <LogControls
                query={testQuery}
                isLive={testIsLive}
                onLiveSelected={handleLive}
                onPauseSelected={handlePause}
                onQueryChanged={handleQueryChange} />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });
});