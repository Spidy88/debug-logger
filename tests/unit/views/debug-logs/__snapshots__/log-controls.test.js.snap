// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogControls> should match its snapshot 1`] = `
Array [
  <div
    className="ðŸ“¦ml_0-75rem ðŸ“¦mr_0-75rem ðŸ“¦mb_1rem ðŸ“¦mt_1rem ðŸ“¦bg_11ev722 ðŸ“¦box-szg_border-box"
  >
    <div
      className="ðŸ“¦dspl_flex ðŸ“¦mr_-1px ðŸ“¦h_40px ðŸ“¦flx-srnk_1 ðŸ“¦flx-grow_0 ðŸ“¦w_10rem ðŸ“¦box-szg_border-box"
    >
      <div
        className="css-3grw6g css-4b8qak ðŸ“¦btlr_3px ðŸ“¦bblr_3px ðŸ“¦box-szg_border-box"
        data-active={true}
      >
        <input
          checked={true}
          className="css-87b09r"
          id="SegmentedControl-60"
          name="SegmentedControl-6"
          onChange={[Function]}
          type="radio"
          value="live"
        />
        <label
          className="css-17kak1e ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦crsr_pointer ðŸ“¦box-szg_border-box"
          htmlFor="SegmentedControl-60"
        >
          Live
        </label>
      </div>
      <div
        className="css-3grw6g css-4b8qak ðŸ“¦btrr_3px ðŸ“¦bbrr_3px ðŸ“¦box-szg_border-box"
        data-active={false}
      >
        <input
          checked={false}
          className="css-87b09r"
          id="SegmentedControl-61"
          name="SegmentedControl-6"
          onChange={[Function]}
          type="radio"
          value="pause"
        />
        <label
          className="css-17kak1e ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦crsr_pointer ðŸ“¦box-szg_border-box"
          htmlFor="SegmentedControl-61"
        >
          Pause
        </label>
      </div>
    </div>
  </div>,
  <div
    className="ðŸ“¦ml_0-75rem ðŸ“¦mr_0-75rem ðŸ“¦mb_1rem ðŸ“¦mt_1rem ðŸ“¦w_100prcnt ðŸ“¦box-szg_border-box"
  >
    <div
      className="ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦h_40px ðŸ“¦flx-grow_1 ðŸ“¦w_100prcnt ðŸ“¦box-szg_border-box"
    >
      <div
        className="ðŸ“¦h_40px ðŸ“¦w_40px ðŸ“¦ptr-evts_none ðŸ“¦pst_absolute ðŸ“¦dspl_flex ðŸ“¦just-cnt_center ðŸ“¦algn-itms_center ðŸ“¦box-szg_border-box"
      >
        <svg
          className="ðŸ“¦z-idx_3 ðŸ“¦w_16px ðŸ“¦h_16px ðŸ“¦box-szg_border-box"
          data-icon="search"
          style={
            Object {
              "fill": "#66788A",
            }
          }
          viewBox="0 0 16 16"
        >
          <title>
            search
          </title>
          <path
            d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 0 0 1.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 1 0 2.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <input
        aria-invalid={false}
        className="css-5ljhhe ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦w_100prcnt ðŸ“¦h_40px ðŸ“¦pl_40px ðŸ“¦pr_13px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦box-szg_border-box"
        disabled={false}
        onChange={[Function]}
        placeholder="Type to search..."
        spellCheck={true}
        test-id="logSearch"
        type="text"
        value=""
      />
    </div>
  </div>,
]
`;
